pkg/smokescreen/config.go

```diff
@@ -79,6 +79,10 @@ type Config struct {
 	// ranges by default (exempting loopback and unicast ranges)
 	// This setting can be used to configure Smokescreen with a blocklist, rather than an allowlist
 	UnsafeAllowPrivateRanges bool
+
+	// Customer handler for verifying requests, users can pass in custom methods to verify requests based
+	// on headers etc.
+	VerifyRequestHandler func(*http.Request) error
```

Comment:
(optional suggestion) I wonder if we want to make it general (eg. `CustomRequestHandler` or `RequestHook`) so it can be used for different use cases. I.e. some folks may use it for verification, others may pass additional data for metrics, etc.
```suggestion
	CustomRequestHandler func(*http.Request) error
```